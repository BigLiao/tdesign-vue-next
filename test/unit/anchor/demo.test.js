import { mount } from '@vue/test-utils';
import base from '@/examples/anchor/demos/base.vue';
import multiple from '@/examples/anchor/demos/multiple.vue';
import target from '@/examples/anchor/demos/target.vue';

// unit test for component in examples.

const props = {
  $route: {
    path: 'http://example.com',
  },
};

describe('Anchor', () => {
  it('base demo works fine', () => {
    const wrapper = mount(base, props);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('bounds demo works fine', () => {
    const wrapper = mount(multiple, props);
    expect(wrapper.element).toMatchSnapshot();
  });
  it('target demo works fine', () => {
    const wrapper = mount(target, props);
    expect(wrapper.element).toMatchSnapshot();
  });
});
