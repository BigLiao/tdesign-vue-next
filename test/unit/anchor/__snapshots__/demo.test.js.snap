// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anchor base demo works fine 1`] = `
<div
  class="anchor-demo"
  id="#default"
  style="display: flex;"
>
  <div
    style="flex: 1;"
  />
  <div>
    
    <div
      class="t-anchor t-size-m"
    >
      <div
        class="t-anchor_line"
      >
        <div
          class="cursor-wrapper"
        >
          <div
            class="cursor"
          />
        </div>
      </div>
      
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="#基础锚点"
          target="_self"
          title="基础锚点"
        >
          基础锚点
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="#多级锚点"
          target="_self"
          title="多级锚点"
        >
          多级锚点
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="#指定容器锚点"
          target="_self"
          title="指定容器锚点"
        >
          指定容器锚点
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="#特定交互锚点"
          target="_self"
          title="特定交互锚点"
        >
          特定交互锚点
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="#尺寸"
          target="_self"
          title="尺寸"
        >
          尺寸
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="##anchor-target-6"
          target="_self"
          title="保密协议"
        >
          保密协议
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="##anchor-target-7"
          target="_self"
          title="其他"
        >
          其他
        </a>
        <!---->
      </div>
      
    </div>
    
  </div>
</div>
`;

exports[`Anchor bounds demo works fine 1`] = `
<div
  class="anchor-demo"
  id="#multiple"
>
  <div
    class="t-anchor t-size-m"
  >
    <div
      class="t-anchor_line"
    >
      <div
        class="cursor-wrapper"
      >
        <div
          class="cursor"
        />
      </div>
    </div>
    
    <div
      class="t-anchor-item"
    >
      <a
        class="t-anchor-item_link"
        href="##default"
        target="_self"
        title="定义"
      >
        定义
      </a>
      <!---->
    </div>
    <div
      class="t-anchor-item"
    >
      <a
        class="t-anchor-item_link"
        href="##multiple"
        target="_self"
        title="服务功能"
      >
        服务功能
      </a>
      <!---->
    </div>
    <div
      class="t-anchor-item"
    >
      <a
        class="t-anchor-item_link"
        href="##small"
        target="_self"
        title="使用指南"
      >
        使用指南
      </a>
      
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="##large"
          target="_self"
          title="创建签名"
        >
          创建签名
        </a>
        <!---->
      </div>
      <div
        class="t-anchor-item"
      >
        <a
          class="t-anchor-item_link"
          href="##container"
          target="_self"
          title="创建内容"
        >
          创建内容
        </a>
        <!---->
      </div>
      
    </div>
    <div
      class="t-anchor-item"
    >
      <a
        class="t-anchor-item_link"
        href="##anchor-target-6"
        target="_self"
        title="保密协议"
      >
        保密协议
      </a>
      <!---->
    </div>
    <div
      class="t-anchor-item"
    >
      <a
        class="t-anchor-item_link"
        href="##anchor-target-7"
        target="_self"
        title="其他"
      >
        其他
      </a>
      <!---->
    </div>
    
  </div>
</div>
`;

exports[`Anchor target demo works fine 1`] = `
<div
  class="anchor-demo"
  id="#anchor-target"
>
  <div
    class="t-anchor-target"
    id="#anchor-target-1"
  >
    
     定义 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-2"
  >
    
     服务功能 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-3"
  >
    
     使用指南 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-4"
  >
    
     创建签名 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-5"
  >
    
     创建内容 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-6"
  >
    
     保密协议 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
  <div
    class="t-anchor-target"
    id="#anchor-target-7"
  >
    
     其他 
    
    <div
      class="t-popup-reference t-copy"
    >
      <transition-stub>
        <div
          aria-hidden="true"
          class="t-popup"
          role="tooltip"
          style="display: none;"
        >
          <div
            class="t-popup-content t-popup-content--arrow"
            style="max-width: 480px;"
          >
            复制链接
            <div
              class="t-popup__arrow"
            />
          </div>
        </div>
      </transition-stub>
      
      <svg
        class="t-icon t-icon-file-copy"
        fill="none"
        height="1em"
        viewBox="0 0 16 16"
        width="1em"
      >
        <path
          d="M4 1.92C4 1.34 4.52 1 5 1h4.37a1 1 0 01.71.3L13.71 5a1 1 0 01.29.7v6.38c0 .58-.52.92-1 .92H5c-.48 0-1-.34-1-.92V1.92zM5 2v10h8V6.01H9V2H5zm5 .65V5h2.32L10 2.65z"
          fill="currentColor"
          fillOpacity="0.9"
        />
        <path
          d="M2 5v9.01a1 1 0 001 1h8v-1H3V5H2z"
          fill="currentColor"
          fillOpacity="0.9"
        />
      </svg>
      
    </div>
  </div>
</div>
`;
